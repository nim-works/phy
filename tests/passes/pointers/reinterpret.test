discard """
  description: "Make sure `Reinterp` involving pointers is lowered"
"""
(TypeDefs
  (ProcTy (Void)))
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 0) 0 (Locals)
    (List
      (Block (Params)
        (Drop (Reinterp (Ptr) (Int 8) (IntVal 1)))
        (Drop (Reinterp (Int 8) (Ptr) (Nil)))
        (Drop (Reinterp (Ptr) (UInt 8) (IntVal 1)))
        (Drop (Reinterp (UInt 8) (Ptr) (Nil)))
        (Return))))))
