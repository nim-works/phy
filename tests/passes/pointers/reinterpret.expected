;$sexp
(TypeDefs
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 0) 0 (Locals)
    (List
      (Block (Params)
        (Drop
          (Reinterp (UInt 8) (Int 8) (IntVal 1))
        )
        (Drop
          (Reinterp (Int 8) (UInt 8) (IntVal 0))
        )
        (Drop (IntVal 1))
        (Drop (IntVal 0))
        (Return)
      )
    )
  )
)
